packages/sources/finage-test/src/index.ts

```diff
@@ -1,9 +1,66 @@
 import { expose, ServerInstance } from '@chainlink/external-adapter-framework'
+import { IncludesFile } from '@chainlink/external-adapter-framework/adapter'
 import { PriceAdapter } from '@chainlink/external-adapter-framework/adapter'
 import { config } from './config'
 import { commodities, crypto, eod, forex, stock, ukEtf } from './endpoint'
+import { PriceEndpointInputParametersDefinition } from '@chainlink/external-adapter-framework/adapter/price'
+import { AdapterParams } from '@chainlink/external-adapter-framework/adapter/types'
+import {
+  AdapterRequest,
+  AdapterRequestContext,
+  AdapterResponse,
+} from '@chainlink/external-adapter-framework/util'
 
-export const adapter = new PriceAdapter({
+export type PriceAdapterRequest<T> = AdapterRequest<T> & {
+  requestContext: AdapterRequestContext<T> & {
+    priceMeta: {
+      inverse: boolean
+    }
+  }
+}
+
+class FinageAdapter extends PriceAdapter<any> {
+  constructor(
+    params: AdapterParams<any> & {
+      includes?: IncludesFile
+    },
+  ) {
+    super(params)
+
+    if (params.includes) {
+      throw Error(
+        'Includes.json cannot be used for inverse behavior due to alternate implementation. refer https://smartcontract-it.atlassian.net/browse/DF-18867',
+      )
+    }
+  }
+
+  override async handleRequest(
```

Comment:
Rather than override, could we just use the default `PriceAdapter.handleRequest`?

Looks like if we tweaked the constructor we could set the `includes`, and so use the default behaviour

Something like
```
 constructor(
    params: AdapterParams<any> & { includes?: IncludesFile },
  ) {

    if (params.includes) {
      throw Error('...')
    }

    super({...params, includes: []})
  }
```

Reply:
@alec this is not working. 

1. The forex endpoint ends up have two transform requests. forexReqTransformer in endpoint/forex.ts which inverts the pairs and sets priceMeta.inverse to True. Then requestTransform from ea-framework/src/adapter/price.ts/PriceAdapter which due to absence of any information in this.includesMap sets priceMeta.inverse to false. 
2. to make this work we will need to add new pairs on the fly to this.includesMap. I dont think endpoints have exposure to includesMap since it is on Adapter level.

Reply:
Ahh that makes sense - I hadn't noticed that the PriceAdapter constructor was also gating based on `includes`. 

In that case can we just add a comment to this function saying that it is copied from PriceAdapter, with a link to the code?